cmd/modelcmd/base_test.go

```diff
@@ -189,6 +192,42 @@ To access it run 'juju switch bar:admin/badmodel'.`,
 	}
 }
 
+func (s *BaseCommandSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+	return ctrl
+}
+
+func (s *BaseCommandSuite) TestNewAPIRootExternalUser(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	conn := mocks.NewMockConnection(ctrl)
+	apiOpen := func(info *api.Info, opts api.DialOpts) (api.Connection, error) {
+		return conn, nil
+	}
+	externalName := "alastair@external"
+	conn.EXPECT().AuthTag().Return(names.NewUserTag(externalName)).Times(3)
+	conn.EXPECT().APIHostPorts()
+	conn.EXPECT().ServerVersion()
+	conn.EXPECT().Addr()
+	conn.EXPECT().IPAddr()
+	conn.EXPECT().PublicDNSName()
+	conn.EXPECT().ControllerAccess().Times(2)
```

Comment:
`.MinTimes(1)` these I would

Reply:
Since calling `AuthTag` 4 times instead of 3 shouldn't be considered a failure imo

Reply:
Changed